### PR TITLE
Prevent WritePendingException during delivery of BOSH packets.

### PR DESCRIPTION
**Pull request by sco0ter**
This exception caused serious trouble, e.g.:
- Clients got an invalid/incomplete packet, which breaks them.
- Openfire logged related exceptions like org.jivesoftware.openfire.http.HttpConnectionClosedException: The http connection is no longer available to deliver content

It was caused by calling complete() although the previous write() has not completed (due to its async nature).

I also restructured that the Content-Length header is only set once.

Fixes OF-989.
See also https://community.igniterealtime.org/thread/57622

### CODE REVIEW

File: src/java/org/jivesoftware/openfire/http/HttpBindServlet.java [hunk 1]
```diff
@@ -447,14 +449,22 @@ public WriteListenerImpl(AsyncContext context, byte[] data) {
...
-            Log.trace("Data can be written to [" + remoteAddress + "]");
-
-            // BOSH communication should not use Chunked encoding.
-            // This is prevented by explicitly setting the Content-Length header.
-            context.getResponse().setContentLength(data.length);
-
-            context.getResponse().getOutputStream().write(data);
-            context.complete();
+            // This method may be invoked multiple times and by different threads, e.g. when writing large byte arrays.
+            Log.info("Data can be written to [" + remoteAddress + "]");
```
Comment (dwd) [inline src/java/org/jivesoftware/openfire/http/HttpBindServlet.java]: Should this be info level logging?

**Comment by tevans** [comment]
+1

File: src/java/org/jivesoftware/openfire/http/HttpBindServlet.java (hunk 1 above)
Comment (tevans) [inline src/java/org/jivesoftware/openfire/http/HttpBindServlet.java]: Agree, let's set this back to trace (or debug).

**Comment by sco0ter** [comment]
Change log level back to trace.

**Pull request merged**